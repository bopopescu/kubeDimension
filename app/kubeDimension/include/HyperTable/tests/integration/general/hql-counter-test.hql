drop table if exists CounterTest;
create table CounterTest (a COUNTER);
insert into CounterTest values ("row", "a", "7");
insert into CounterTest values ("row", "a", "5");
insert into CounterTest values ("row", "a", "=4");
insert into CounterTest values ("row", "a", "+3");
insert into CounterTest values ("row", "a", "+7");
insert into CounterTest values ("row", "a", "=19");
insert into CounterTest values ("row", "a", "5");
insert into CounterTest values ("row", "a", "+8");
compact type=minor table CounterTest;
pause 5;
insert into CounterTest values ("row", "a", "+32");
insert into CounterTest values ("row", "a", "=7");
insert into CounterTest values ("row", "a", "9");
select * from CounterTest into file "hql-counter-test.output.1";
compact table CounterTest;
pause 5;
select * from CounterTest into file "hql-counter-test.output.2";