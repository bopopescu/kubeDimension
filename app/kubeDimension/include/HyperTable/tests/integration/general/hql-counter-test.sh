#!/usr/bin/env bash

HT_HOME=${INSTALL_DIR:-"$HOME/hypertable/current"}
SCRIPT_DIR=`dirname $0`

$HT_HOME/bin/ht-start-test-servers.sh --clear --no-thriftbroker \
    --Hypertable.RangeServer.Maintenance.Interval=500

cat $SCRIPT_DIR/hql-counter-test.hql | $HT_HOME/bin/ht shell --batch
if [ $? -ne 0 ]; then
    $HT_HOME/bin/ht-stop-servers.sh
    echo "Hypertable shell failure!"
    exit 1
fi

diff hql-counter-test.output.1 $SCRIPT_DIR/hql-counter-test.golden
if [ $? -ne 0 ]; then
    $HT_HOME/bin/ht-stop-servers.sh
    echo "Diff encountered!"
    exit 1
fi

diff hql-counter-test.output.2 $SCRIPT_DIR/hql-counter-test.golden
if [ $? -ne 0 ]; then
    $HT_HOME/bin/ht-stop-servers.sh
    echo "Diff encountered!"
    exit 1
fi

$HT_HOME/bin/ht-stop-servers.sh
exit 0

