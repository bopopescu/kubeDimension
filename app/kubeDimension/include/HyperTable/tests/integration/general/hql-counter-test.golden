#row	column	value
row	a	16
